python/build_definitions/tcmalloc.py

```diff
@@ -41,7 +41,6 @@ def build(self, builder: BuilderInterface) -> None:
                 dep=self, src_file="tcmalloc_static.a", dest_file="libgoogletcmalloc.a",
                 src_folder="tcmalloc", is_shared=False)
 
-        # Copy headers.
-        builder.log_output(log_prefix, ['mkdir', '-p', builder.prefix_include + '/tcmalloc'])
-        builder.log_output(log_prefix, ['cp'] + glob.glob('./tcmalloc/*.h') +
-                                       [builder.prefix_include + '/tcmalloc'])
+        # Copy headers, keeping the folder structure. https://stackoverflow.com/a/29457076.
+        builder.log_output(log_prefix, ["rsync", "-a", "--include=*.h", "-f",
+                                        "hide,! */", "./tcmalloc", builder.prefix_include])
```

Comment:
I think the best discussion for this is at https://stackoverflow.com/questions/9622883/recursive-copy-of-a-specific-file-type-maintaining-the-file-structure-in-unix-li/9626253#9626253

Also I don't get how "hide" works.

Reply:
Will update that comment to point to that discussion. There's a description of what `-f 'hide,! */'` does on a comment there too. In short, `-f 'hide,! */'` is a filter that excludes all files (`*/` matches all directories and `!` negates it to match all files), and the `--include=*.h` adds back only the headers.